File: java/src/test/java/io/cucumber/RunnerTest.java
package io.cucumber;

import gherkin.AstBuilder;
import gherkin.Parser;
import gherkin.ast.GherkinDocument;
import gherkin.pickles.Compiler;
import gherkin.pickles.Pickle;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;
import java.util.regex.Pattern;

import static org.junit.Assert.assertEquals;


public class RunnerTest {

    @Test
    public void itRecordsResults() {
        Parser<GherkinDocument> parser = new Parser<>(new AstBuilder());
        Compiler compiler = new Compiler();

        String feature = String.join("\n",
                "Feature:",
                "  Scenario:",
                "    Given a passing step"
                );
        GherkinDocument gherkinDocument = parser.parse(feature);

        List<Pickle> pickles = compiler.compile(gherkinDocument, "path/to/the.feature");

        List<StepDefinition> stepDefinitions;
        stepDefinitions = Arrays.asList(
                new StepDefinition(Pattern.compile("pass"), () -> {})
        );

        Glue glue = new Glue(stepDefinitions);
        Runner runner = new Runner(glue);

        Report report = runner.execute(pickles);

        assertEquals(1, report.testCases.size());
        assertEquals(1, report.testCasesPassed.size());
    }
}

File: dotnet/MicroCuke/MicroCuke.Tests/RunnerTest.cs
using System;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Gherkin;
using System.Collections.Generic;
using System.Text.RegularExpressions;
using Gherkin.Ast;
using System.IO;

namespace MicroCuke.Tests
{
  [TestClass]
  public class RunnerTest
  {
    [TestMethod]
    public void ItRecordsResults()
    {
      var parser = new Parser();
      String feature = @"
                Feature:
                  Scenario:
                    Given a passing step";

      var gherkinDocument = parser.Parse(new StringReader(feature));

      // No Pickles for Gherkin.NET yet
      // so we'll just use the first scenario, and assume there are
      // no backgrounds or scenario outlines

      var scenario = gherkinDocument.Feature.Children.First();

      var stepDefinitions = new List<StepDefinition>()
      {
        new StepDefinition(new Regex("pass"), () => {})
      };

      var glue = new Glue(stepDefinitions);
      var runner = new Runner(glue);

      var report = runner.Execute(new List<ScenarioDefinition>() { scenario });

      Assert.AreEqual(1, report.TestCases.Count);
      Assert.AreEqual(1, report.TestCasesPassed.Count);
    }
  }
}
